airflow/jobs/scheduler_job.py

```diff
@@ -750,7 +600,16 @@ def _process_dags(self, dags: List[DAG], session=None):
             if dag_runs_for_dag:
                 tis_out.extend(self._process_task_instances(dag, dag_runs_for_dag))
                 if check_slas:
-                    self.manage_slas(dag)
+                    # Look for SLA misses for all task instances associated with this
+                    # DAG, whether or not corresponding runs have been created yet.
+                    # Catch exceptions, though; we don't want SLA checking failures to
+                    # prevent tasks from being scheduled!
+                    try:
+                        dag.manage_slas()
+                    except Exception:  # pylint: disable=broad-except
```

Comment:
since we will be using sla for critical monitoring, i think we should do more than just log the exception. how about adding a metric for this so one can setup alerting for issues that impacts sla monitoring?

Reply:
good idea!